src/NumericalAlgorithms/Interpolation/IrregularInterpolant.hpp

```diff
@@ -0,0 +1,104 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Declares functions for interpolating variables to a set of points.
+
+#pragma once
+
+#include <cstddef>
+#include <ostream>
+
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Blas.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Index;
+namespace PUP {
+class er;
+}  // namespace PUP
+// IWYU pragma: no_forward_declare Variables
+/// \endcond
+
+/// \ingroup NumericalAlgorithmsGroup
+/// Interpolates a `Variables` onto an arbitrary set of points.
+template <size_t Dim>
+class IrregularInterpolant {
+ public:
+  IrregularInterpolant(const Index<Dim>& mesh,
+                       const tnsr::I<DataVector, Dim>& points) noexcept;
```

Comment:
What frame are these points in?

Reply:
Logical.  Changing this now.  Unfortunately I need to jump through hoops to get MathFunctions::TensorProduct to work with a frame other than Inertial :(